lib/chef-dk/command/generator_commands.rb

```diff
@@ -37,19 +37,22 @@ module SharedGeneratorOptions
         :short => "-I LICENSE",
         :long => "--license LICENSE",
         :description => "all_rights, apache2, mit, gplv2, gplv3 - defaults to all_rights",
-        :default => "all_rights"
+        :default => "all_rights",
```

Comment:
The proc options in mixlib-cli are evaluated when the option is passed via the CLI, so the code:

```ruby
:proc => Proc.new { |key| Chef::Config[:knife][:cookbook_license] = key }
```

means that `chef generate THING -l apache2` will cause mixlib-cli to call that proc with "apache2" as the argument, which results in `Chef::Config[:knife][:cookbook_license] = "apache2"`. So if you access the `cookbook_license` setting via `Chef::Config[:knife][:cookbook_license]`, you'll end up with the value set on the command line. However, just doing that is not enough. When I pulled this down and set my email in my configfile, this change didn't result in the generator actually using it.

mixlib-cli provides the command line options as a Hash that you access via the `config` method. ChefDK adds those to the generator's context in a loop here: https://github.com/chef/chef-dk/blob/b227c377703625ba5da2744d5dbc623c6ba923be/lib/chef-dk/command/generator_commands/base.rb#L73-L77 What I think you need to do is to merge the config file data with the mixlib-cli config. However, this is a bit tricky because we need to only overwrite what's in `config` if it's a default value (we want default < config file < cli). So we need to use the `use_separate_default_options` feature of mixlib-cli: https://github.com/chef/mixlib-cli/blob/13663ac9759ce5f1a0c9af68ec78028586a7e1e5/lib/mixlib/cli.rb#L110-L129 and then we should be able to correctly merge the 3 sources of configuration.